run.sh

```diff
@@ -42,6 +42,34 @@ fi
 
 # ========================================================================
 
+# Variable to store whether --skip was used in parameters
```

Comment:
We're not going to move the full cloning logic above the `--skip` check because it can take a while.

Just this part:

```
# Check if the commit hash argument is provided
if [ -z "$1" ]; then
    echo "Please provide the commit hash or tag as an argument."
    exit 1
fi
```